### PR TITLE
Builder Constructor examples

### PR DESCRIPTION
**Pull request by jesuspedge**
## Description

This PR added two examples using the builder constructor.

## Examples

- [x] CDs example
- [x] Blue Rays example

## Demo Video

https://github.com/the-d-velopers/animated_book_list/assets/74924269/281ffc23-af9a-4bd3-abe2-9a68de93acfd

### CODE REVIEW

File: example/lib/examples/builder_constructor_cds_example.dart [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import 'dart:math' as math;
+
+import 'package:animated_book_list/animated_book_list.dart';
+import 'package:flutter/material.dart';
+
+///Creating a class object
+class Cd {
+  String cdCoverImgUrl;
+  String cdDiscImgUrl;
+  List<Color> cdGradient;
```
Comment (victormanuelfrancodev) [inline example/lib/examples/builder_constructor_cds_example.dart]: try to use inmutable variables 
```
final String cdCoverImgUrl;
...
...
```

File: example/lib/examples/builder_constructor_cds_example.dart [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+      ///Number of items on the list
+      itemCount: cds.length,
+
+      //Item constructor
+      itemBuilder: (_, index) {
+        ///For each object we return a widget constructor with the data.
+        return AnimatedBookWidget.builder(
+          ///Size parameter
+          size: horizontalView ? Size.fromWidth(225) : Size.fromHeight(225),
```
Comment (victormanuelfrancodev) [inline example/lib/examples/builder_constructor_cds_example.dart]: Values such as the size of the widgets (Size.fromWidth(225), Size.fromHeight(225)) could be defined as constants at the beginning of your file to facilitate global adjustments and improve readability.

**Review by victormanuelfrancodev — CHANGES_REQUESTED** [review]

File: example/lib/examples/builder_constructor_cds_example.dart (hunk 2 above)
Comment (jesuspedge) [reply example/lib/examples/builder_constructor_cds_example.dart]: Correct!

File: example/lib/examples/builder_constructor_brays_example.dart [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+          ///Creates the child inside the book
+          contentChild: Padding(
+            padding: const EdgeInsets.all(5.0),
+            child: Image.network(bRays[index].bRayDiscImgUrl),
+          ),
+
+          ///Customize the back content  and use the animation based on open cover animation
+          contentBuilder: (context, cdAnimation, child) {
+            return Container(
```
Comment (monster555) [inline example/lib/examples/builder_constructor_brays_example.dart]: You are using only `decoration`. Prefer to use `DecoratedBox` over `Container`

File: example/lib/examples/builder_constructor_cds_example.dart [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+          ///Creates the child inside the book
+          contentChild: Padding(
+            padding: const EdgeInsets.all(15.0),
+            child: Image.network(cds[index].cdDiscImgUrl),
+          ),
+
+          ///Customize the back content  and use the animation based on open cover animation
+          contentBuilder: (context, cdAnimation, child) {
+            return Container(
```
Comment (monster555) [inline example/lib/examples/builder_constructor_cds_example.dart]: You are using only `decoration`. Prefer to use `DecoratedBox` over `Container`

File: example/lib/home_page.dart [hunk 5]
```diff
@@ -88,7 +88,37 @@ class _MyHomePageState extends State<MyHomePage> {
...
             Text(
-              'Separated example:',
+              horizontalView
+                  ? 'Builder Cds example:'
+                  : 'Builder\nCds\nexample:',
+              style: Theme.of(context).textTheme.headlineSmall,
+            ),
+            Padding(
+              padding: const EdgeInsets.all(10),
+              child: SizedBox(
```
Comment (monster555) [inline example/lib/home_page.dart]: When `width` and `height` are the same, prefer to use the `SizedBox.square` constructor

**Review by monster555 — CHANGES_REQUESTED** [review]
I'd suggest considering an 8pt or 4pt grid for paddings and widget sizes to ensure a visually cohesive design

**Review by br-programmer — APPROVED** [review]

**Review by diegoveloper — APPROVED** [review]

**Pull request merged**